app/src/main/jni/Application.mk

```diff
@@ -4,7 +4,7 @@ APP_STL := gnustl_static
 APP_ABI := armeabi-v7a x86 arm64-v8a
 
 # For releases
-APP_CFLAGS := -Ofast -fmodulo-sched -fmodulo-sched-allow-regmoves -fsingle-precision-constant -fvariable-expansion-in-unroller -fomit-frame-pointer -fno-math-errno -funsafe-math-optimizations -ffinite-math-only -fdata-sections -fbranch-target-load-optimize2 -fno-exceptions -fno-stack-protector -fforce-addr -ftree-loop-im -ftree-loop-ivcanon -fivopts -ftree-loop-if-convert-stores -floop-strip-mine -ftree-loop-distribution -floop-interchange -ftree-loop-linear -floop-block -Wno-psabi
+APP_CFLAGS := -O2 -fmodulo-sched -fmodulo-sched-allow-regmoves -fsingle-precision-constant -fvariable-expansion-in-unroller -fomit-frame-pointer -fno-math-errno -funsafe-math-optimizations -ffinite-math-only -fdata-sections -fbranch-target-load-optimize2 -fno-exceptions -fno-stack-protector -fforce-addr -ftree-loop-im -ftree-loop-ivcanon -fivopts -ftree-loop-if-convert-stores -floop-strip-mine -ftree-loop-distribution -floop-interchange -ftree-loop-linear -floop-block -Wno-psabi

```

Comment:
Ofast is basically O3 plus a few other things. O3 is actually a very dangerous optimization level, and even though it might bring better performance, it can also break the code.
O2 is the recommended level of optimization without risking breakage, according to the GCC manual.